cmd/setup.go

```diff
@@ -304,3 +308,39 @@ func configureLoadPoints(conf config, cp *ConfigProvider) (loadPoints []*core.Lo
 
 	return loadPoints, nil
 }
+
+func configureCircuits(site *core.Site, loadPoints []*core.LoadPoint, cp *ConfigProvider) (err error) {
+	ccInterfaces, _ := viper.AllSettings()["circuits"].([]interface{})
+
+	for _, ccI := range ccInterfaces {
+		var ccMap map[string]interface{}
+		if err := util.DecodeOther(ccI, &ccMap); err != nil {
+			return fmt.Errorf("failed decoding circuit configuration: %w", err)
+		}
+
+		ccNew, err := core.NewCircuitFromConfig(cp, ccMap, site)
+		if err != nil {
+			return fmt.Errorf("failed configuring circuit: %w", err)
+		}
+		site.Circuits = append(site.Circuits, ccNew)
+	}
+	// connect circuits and lps
+	for lpId := range loadPoints {
+		for ccId := range site.Circuits {
+			loadPoints[lpId].CircuitPtr = site.Circuits[ccId].GetCircuit(loadPoints[lpId].CircuitRef)
```

Comment:
was passiert wenn GetCircuit nix findet? Fehlerhandlung!

Reply:
passiert direkt drunter: `if loadPoints[lpId].CircuitRef != "" && loadPoints[lpId].CircuitPtr == nil`
Es muss nicht zwingend einen Toplevel Circuit geben, daher muss in alle Circuits geschaut werden bevor ein Fehler kommt